src/config.ts

```diff
@@ -22,13 +22,19 @@ export class AgentConfig {
     write(this.configuration, this.configFile);
   }
 
-  public getMQTTUrl(): string {
+  public getWebsocketURL(): string {
     if (
       this.configuration.mqttProtocol === "ws" ||
       this.configuration.mqttProtocol === "wss"
     ) {
-      return `${this.configuration.mqttProtocol}://${this.configuration.domain}:${this.configuration.mqttPort}/mqtt`;
+      return `${this.configuration.mqttProtocol}://${this.configuration.domain}:${this.configuration.mqttPort}`;
     }
+
+    // mqttProtocol is not a WebSocket prefix. Then fall back to the secure connection.
+    return `wss://${this.configuration.domain}:${this.configuration.mqttPort}`;
+  }
+
+  public getMQTTUrl(): string {
```

Comment:
I guess the old implementation of `getMQTTUrl` did not actually work for mqtt/mqtts protocols (none websocket) as the `/mqtt` path was always used, but probably should not have been used for mqtt(s).. Maybe you could adjust this?

Otherwise this looks good :+1: